pkg/kubelet/rkt/rkt.go

```diff
@@ -425,6 +436,65 @@ func (r *Runtime) makePodManifest(pod *api.Pod, pullSecrets []api.Secret) (*appc
 	var globalPortMappings []kubecontainer.PortMapping
 	manifest := appcschema.BlankPodManifest()
 
+	listReq := &rktapi.ListPodsRequest{
+		Filter: &rktapi.PodFilter{
+			States: []rktapi.PodState{
+				//TODO: In the future some pods can remain running after some apps exit: https://github.com/appc/spec/pull/500
+				rktapi.PodState_POD_STATE_RUNNING,
+				rktapi.PodState_POD_STATE_EXITED,
+				rktapi.PodState_POD_STATE_DELETING,
+				rktapi.PodState_POD_STATE_GARBAGE,
+			},
+			Annotations: []*rktapi.KeyValue{
+				&rktapi.KeyValue{
+					Key:   k8sRktKubeletAnno,
+					Value: k8sRktKubeletAnnoValue,
+				},
+				&rktapi.KeyValue{
+					Key:   k8sRktUIDAnno,
+					Value: string(pod.UID),
+				},
+			},
+		},
+	}
+	listResp, err := r.apisvc.ListPods(context.Background(), listReq)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't list pods: %v", err)
+	}
+
+	restartCount := 0
+	for _, rktpod := range listResp.Pods {
+        //TODO: get the manifest from listresp.Pods when this gets merged: https://github.com/coreos/rkt/pull/1786
+        inspectReq := &rktapi.InspectPodRequest{rktpod.Id}
+        inspectResp, err := r.apisvc.InspectPod(context.Background(), inspectReq)
+        if err != nil {
+            return nil, err
+        }
+
+		manifest := appcschema.PodManifest{}
+		err = json.Unmarshal(inspectResp.Pod.Manifest, manifest)
+		if err != nil {
+			return nil, err
+		}
+
+		if countString, ok := manifest.Annotations.Get(k8sRktRestartCountAnno); ok {
+			num, err := strconv.Atoi(countString)
+			if err != nil {
+				continue
```

Comment:
I had a feeling that if I errored on that, that one could prevent the kubelet from restarting a given pod by inserting a pod with a matching uid annotation and a garbage restart count annotation.